File: src/api/Models/Survey.cs
using System.Text.Json.Nodes;

namespace Trey.Api.Models;

public enum SurveyLanguage
{
    En,
    Fi
}
public class Survey
{
    public Guid? Id { get; set; }
    public TimeRange? ResponsePeriod { get; set; }
    // <lang, name>
    public Dictionary<SurveyLanguage, string> Name { get; set; } = new();
    public JsonObject? SurveyJson { get; set; }
}

public class SurveyAnswer
{
    public Guid? Id { get; set; }
    public Guid SurveyId { get; set; }
    public string? OrganizationId { get; set; }
    public string? AnswerJson { get; set; }
}

File: src/web/src/models/survey.ts
export enum SurveyLanguage {
  En = "En",
  Fi = "Fi",
}

interface TimeRange {
  start?: Date
  end?: Date
}

export enum SurveyType {
  AssociationAnnouncement,
  OperationalGrant,
  Other,
}

export interface Survey {
  id?: string
  name: Record<SurveyLanguage, string>
  surveyJson?: string
  surveyType: SurveyType
  responsePeriod?: TimeRange
}

export interface SurveyAnswer {
  id?: string
  surveyId: string
  organizationId?: string
  answerJson?: string
  updatedAt?: Date
}

export interface CachedSurveyState {
  surveyId: string
  organizationId?: string
  answerJson: string // JSON string
  updatedAt: string // ISO timestamp
}
